tests/UKHO.S100PermitService.Common.UnitTests/IO/PermitReaderWriterTests.cs

```diff
@@ -1,34 +1,89 @@
+            var result = _permitReaderWriter.ReadXsdVersion();
+
+            result.Should().NotBeNullOrEmpty();
         }
 
         [Test]
         public void WhenProductModelIsPassed_ThenReturnXmlString()
         {
-            var expectedResult = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n<Permit xmlns:S100SE=\"http://www.iho.int/s100/se/5.2\" xmlns:ns2=\"http://standards.iso.org/iso/19115/-3/gco/1.0\" xmlns=\"http://www.iho.int/s100/se/5.2\">\r\n  <S100SE:header>\r\n    <S100SE:issueDate>2024-09-02+01:00</S100SE:issueDate>\r\n    <S100SE:dataServerName>fakeDataServerName</S100SE:dataServerName>\r\n    <S100SE:dataServerIdentifier>fakeDataServerIdentifier</S100SE:dataServerIdentifier>\r\n    <S100SE:version>1</S100SE:version>\r\n    <S100SE:userpermit>fakeUserPermit</S100SE:userpermit>\r\n  </S100SE:header>\r\n  <S100SE:products>\r\n    <S100SE:product id=\"fakeID\">\r\n      <S100SE:datasetPermit>\r\n        <S100SE:filename>fakefilename</S100SE:filename>\r\n        <S100SE:editionNumber>1</S100SE:editionNumber>\r\n        <S100SE:expiry>2024-09-02</S100SE:expiry>\r\n        <S100SE:encryptedKey>fakeencryptedkey</S100SE:encryptedKey>\r\n      </S100SE:datasetPermit>\r\n    </S100SE:product>\r\n  </S100SE:products>\r\n</Permit>";
-            var products = new List<Products>()
+            A.CallTo(() => _fakeSchemaValidator.ValidateSchema(A<string>.Ignored, A<string>.Ignored)).Returns(true);
+
+            var result = _permitReaderWriter.CreatePermits(GetPermitDetails());
+
+            result.Should().NotBeNull();
+
+            var stringResult = ConvertMemoryStreamToXmlString(result);
+            var trimmedResult = TrimXml().Replace(stringResult, string.Empty);
+            trimmedResult.Should().Be(GetExpectedXmlString());
+        }
+
+        [Test]
+        public void WhenSchemaIsInvalid_ThenReturnsFalse()
+        {
+            A.CallTo(() => _fakeSchemaValidator.ValidateSchema(A<string>.Ignored, A<string>.Ignored)).Returns(false);
+
+            FluentActions.Invoking(() => _permitReaderWriter.CreatePermits(GetInValidPermitDetails())).Should().
+                                            ThrowExactly<PermitServiceException>().WithMessage("Invalid permit xml schema");
+        }
+
+        private Dictionary<string, Permit> GetPermitDetails()
+        {
+            var fakePermitDictionary = new Dictionary<string, Permit>();
+
+            var fakeProducts = new List<Products>()
             {
                 new()
                 {
-                     Id = "fakeID",
+                     Id = "fakeID1",
+                     DatasetPermit =
+                     [
+                         new ProductsProductDatasetPermit()
+                         {
+                             EditionNumber = 1,
+                             EncryptedKey = "fakeencryptedkey",
+                             Expiry = DateTime.Parse("2024-09-02"),
+                             Filename = "fakefilename",
```

Comment:
line no. 94 & 147 as well remove , from end and please add one blank line after line no.98
```suggestion
                             Filename = "fakefilename"
```

Reply:
Resolved